wicketd/src/update_tracker.rs

```diff
@@ -1560,14 +1573,24 @@ impl UpdateContext {
         cx: StepContext<S>,
         stage: ComponentUpdateStage,
         update_id: Uuid,
-        component: &str,
+        component: UpdateComponent,
+        component_name: &str,
     ) -> anyhow::Result<()>
     where
         S::ProgressMetadata: Default,
     {
         // How often we poll MGS for the progress of an update once it starts.
         const STATUS_POLL_FREQ: Duration = Duration::from_millis(300);
 
+        let progress_units = match component {
+            UpdateComponent::Rot => {
+                // XXX what's the correct unit for this?
+                ProgressUnits::new("flash slots scanned")
+            }
+            UpdateComponent::Sp => ProgressUnits::new("flash slots scanned"),
```

Comment:
Ug, with apologies, this isn't quite right, and I'm not sure how to suggest making this right. There are two issues at play:

1. The only SP updates that deliver "preparing" status are sidecar SP updates - neither gimlet nor psc SPs should ever return a preparing status, and if they do at some point in the future there's no guarantee this would be the correct unit.
2. For sidecar, preparation is divided into two stages: first, the SP has to scan the flash slots, then if it doesn't find a match, it has to erase one of the slots. When the SP [sends its (unitless) progress](https://github.com/oxidecomputer/hubris/blob/master/task/control-plane-agent/src/update/sp/auxflash.rs#L88-L103), the total count is "number of slots + number of sectors to erase", where a `current` value less than "number of slots" means it's still scanning and a `current` value >= the number of slots means it's erasing (and if erasing is unnecessary, it will skip from "number of slots" directly to "preparation complete").

Would it be unreasonable to lean into the unitless idea here and make both this and the RoT something like `"preparation steps"`? It's not particularly useful, but I'm not sure there _is_ much use to anyone other than a hubris developer in the specific units here.

Reply:
Yeah, that makes sense. I'll change this to "preparation steps".

Reply:
Done.